### PR TITLE
finish chap 12

### PR DESCRIPTION
**Pull request by nguyenthethang1995**

### CODE REVIEW

**Comment by nguyenthethang1995** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -46,6 +46,20 @@ def send_activation_email
     UserMailer.account_activation(self).deliver_now
   end
 
+  def create_reset_digest
+    self.reset_token = User.new_token
+    update_attributes reset_digest: User.digest(reset_token)
+    update_attributes reset_sent_at: Time.zone.now
```
Comment (huongnguyenmta) [inline app/models/user.rb]: `update_attributes reset_digest: User.digest(reset_token), reset_sent_at: Time.zone.now`

**Comment by huongnguyenmta** [comment]
commented

**Comment by nguyenthethang1995** [comment]
ready

**Comment by huongnguyenmta** [comment]
merged
Mớt nha (nod)

**Pull request merged**